modules/decklinksrc/decklinksrc.py

```diff
@@ -1,40 +1,42 @@
 import cv2 as cv2
 import numpy as np
+import threading
+
 
 class DeckLinkSRC:
 
-    def __init__(self):
+    def __init__(self, videoPipeline):
         self.__currentFrame = None
-        self.capture = cv2.VideoCapture('decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink') #Starts capture on initialization of object
-        #Because of this, we no longer need start stream code from Aryan
+        self.capture = cv2.VideoCapture(
+            'decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink')  # Starts capture on initialization of object
+        self.videoPipeline = videoPipeline
+        main = threading.Thread(target=self._main_)
+        main.start()
+
+    def _main_(self):
+        self.videoPipeline.addNewPackage(self.grab())
 
-    def stop(self): #Logic for stopping video feed by releasing capture and destroying any windows open.
+    def stop(self):  # Logic for stopping video feed by releasing capture and destroying any windows open.
         self.capture.release()
         cv2.destroyAllWindows()
-    
 
-    def grab(self, frame): #Logic for grabbing frame from deckLink
-        self.cv2.imshow('Grabbedframe', frame)
-    
+    def grab(self):  # Logic for displaying single frame from deckLink
+        self.__currentFrame = self.capture.read()
+        return self.get_frame()
```

Comment:
This function isn't really displaying the image like the old one was, it seems to just be retuning the video frame data

Reply:
This issue was fixed with the DeckLinkSRC merge (merge that in before final PR)

Reply:
Done